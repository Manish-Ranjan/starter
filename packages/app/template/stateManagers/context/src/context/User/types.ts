export enum UserActionTypes {
    ADD_USER = 'ADD_USER',
    REMOVE_USER = 'REMOVE_USER'
}

export type User = {
    readonly firstName: string;
    readonly lastName: string;
    readonly email: string;
    readonly phoneNumber: string;
};

export type AddUserAction = User & {
    type: typeof UserActionTypes.ADD_USER;
};

export type RemoveUserAction = {
    type: typeof UserActionTypes.REMOVE_USER;
};

export type UserActions = AddUserAction | RemoveUserAction;

export type Dispatch = (action: UserActions) => void;
