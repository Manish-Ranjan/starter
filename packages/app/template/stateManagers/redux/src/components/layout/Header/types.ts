import { fetchUser } from '@store/user';

export type HeaderStateProps = {
    firstName: string;
    lastName: string;
};

export type HeaderDispatchProps = {
    fetchUser: typeof fetchUser;
};

export type HeaderProps = HeaderStateProps & HeaderDispatchProps;
