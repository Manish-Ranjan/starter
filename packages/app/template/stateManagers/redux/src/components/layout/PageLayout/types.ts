import { fetchUser, User } from '@store/user';
import { HTMLProps } from '@medly-components/utils';

export type StateProps = {
    currentUser: User;
};

export type DispatchProps = {
    fetchUser: typeof fetchUser;
};

export type Props = StateProps & DispatchProps & HTMLProps<HTMLDivElement>;
