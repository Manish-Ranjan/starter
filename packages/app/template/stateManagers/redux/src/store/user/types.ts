import { Action } from 'redux';

export enum UserActionTypes {
    ADD_USER = '@@user/ADD_USER',
    REMOVE_USER = '@@user/REMOVE_USER'
}

export interface UserState {
    userName: string;
    email: string;
    phoneNumber: string;
}

export interface AddUserAction extends Action {
    userName: string;
    email: string;
    phoneNumber: string;
    type: typeof UserActionTypes.ADD_USER;
}

export interface RemoveUserAction extends Action {
    type: typeof UserActionTypes.REMOVE_USER;
}

export type UserActions = AddUserAction | RemoveUserAction;
