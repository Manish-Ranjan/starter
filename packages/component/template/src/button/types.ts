import { HTMLProps } from '@medly-components/utils';

export type ButtonProps = HTMLProps<HTMLButtonElement> & {
    /** Button design */
    variant: 'solid' | 'outlined';
    /** Button type */
    type?: 'button' | 'reset' | 'submit';
};
