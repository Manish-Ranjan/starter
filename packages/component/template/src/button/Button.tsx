import type { FC } from 'react';
import { ButtonStyled } from './Button.styled';
import { ButtonProps } from './types';

export const Button: FC<ButtonProps> = props => <ButtonStyled {...props} />;
Button.defaultProps = {
    variant: 'solid'
};
