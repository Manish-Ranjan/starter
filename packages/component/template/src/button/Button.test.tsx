import { render } from '@testing-library/react';
import React from 'react';
import { Button } from './Button';
import { ButtonProps } from './types';

describe('Button', () => {
    test.each(['solid', 'outlined'])('should render properly with %p variant', (variant: ButtonProps['variant']) => {
        const { container } = render(<Button variant={variant}>Button</Button>);
        expect(container).toMatchSnapshot();
    });
});
