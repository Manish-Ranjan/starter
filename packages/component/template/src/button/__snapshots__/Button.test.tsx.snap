// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render properly with "outlined" variant 1`] = `
.c0 {
  padding: 10px 20px;
  border-radius: 4px;
  border: 1px solid blue;
  background-color: white;
  color: blue;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;

exports[`Button should render properly with "solid" variant 1`] = `
.c0 {
  padding: 10px 20px;
  border-radius: 4px;
  border: 0;
  background-color: blue;
  color: white;
}

<div>
  <button
    class="c0"
  >
    Button
  </button>
</div>
`;
